src/concepts/kafka-proxies/grpc-proxy.md

```diff
@@ -1,297 +1,61 @@
-  options:
-    host: <grpc_host>
-    port: <grpc_port>
-```
-
-### Proxy service entrypoint
-
-Using a [tcp](../../reference/config/bindings/binding-tcp.md) server binding we can route the gRPC traffic through an `http` server, `<entrypoint_host>`:`<entrypoint_port>`, to our desired gRPC server at the `<grpc_server_binding_name>`. The [tls](../../reference/config/bindings/binding-tls.md) server binding secures the entrypoint by defining a tls certificate from the [filesystem](../../reference/config/vaults/vault-filesystem.md) vault, which is used to send the traffic over https. Alternatively, if tls is not needed The tcp server can exit directly to the http server and the tls binding and filesystem vault can be removed.
-
-```yaml
-  tcp_server:
-    type: tcp
-    kind: server
-    options:
-      host: 0.0.0.0
-      port: <entrypoint_port>
-    exit: tls_server
-  tls_server:
-    type: tls
-    kind: server
-    vault: server
-    options:
-      keys:
-        - <entrypoint_host>
-      sni:
-        - <entrypoint_host>
-      alpn:
-        - h2
-    exit: http_server
-  http_server:
-    type: http
-    kind: server
-    options:
-      versions:
-        - h2
-      access-control:
-        policy: cross-origin
-    routes:
-      - when:
-          - headers:
-              :scheme: https
-              :authority: <entrypoint_host>:<entrypoint_port>
-        exit: <grpc_server_binding_name>
-```
-
-### Connecting to Kafka
-
-A gRPC server can connect to Kafka the same as any other binding in Zilla. [See the docs](../../how-tos/connecting-to-kafka/apache-kafka.md) on the different connection options.
-
-```yaml
-  kafka_cache_client:
-    type: kafka
-    kind: cache_client
-    exit: kafka_cache_server
-  kafka_cache_server:
-    type: kafka
-    kind: cache_server
-    exit: kafka_client
-  kafka_client:
-    type: kafka
-    kind: client
-    exit: tcp_client
-  tcp_client:
-    type: tcp
-    kind: client
-    options:
-      host: <kafka_host>
-      port: <kafka_port>
-    routes:
-      - when:
-          - cidr: 0.0.0.0/0
-```
+Method routes can use [custom metadata fields](../../reference/config/bindings/binding-grpc-kafka.md#when-metadata) for request routing and idempotency. Zilla can also augment the metadata it sends based on the configured route the request matches.
 
-## Try it out
+## Reliable Delivery
 
-Go check out the [grpc.kafka.proxy](https://github.com/aklivity/zilla-examples/tree/main/grpc.kafka.proxy) example for a full implementation of an EchoService.
+Zilla sends an event ID with every message serialized as an unknown field in the payload. Any message can be identified without field collision, and the client doesn't need to acknowledge the message receipt explicitly. A client consuming a stream of messages can remember the event ID. If the event the stream gets interrupted. The client reconnects with a `last-event-id` header to recover without message loss or needing to start over from the beginning.
```

Comment:
I think this last section feels like it is part of the other sections, like you can still do all those things above but also add reliable delivery.

However, this reliably delivery is a oneway broadcast only, so I think we need to make that context more clear before describing the reliability aspects, just to avoid any potential confusion.